src/main/java/com/mraof/minestuck/block/redstone/TimedSolidSwitchBlock.java

```diff
@@ -0,0 +1,125 @@
+package com.mraof.minestuck.block.redstone;
+
+import com.mraof.minestuck.util.ParticlesAroundSolidBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.particles.RedstoneParticleData;
+import net.minecraft.state.IntegerProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.util.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.common.util.Constants;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+/**
+ * When right clicked without crouching(and is unpowered), the block turns to full power and brightness and then loses one value of power and light at a set rate of ticks determined by the "tickrate" of the block
+ */
+public class TimedSolidSwitchBlock extends Block
+{
+	public static final IntegerProperty POWER = BlockStateProperties.POWER;
+	
+	private final int tickRate;
+	
+	public TimedSolidSwitchBlock(Properties properties, int tickRate)
+	{
+		super(properties);
+		this.tickRate = tickRate;
+		registerDefaultState(stateDefinition.any().setValue(POWER, 0));
+	}
+	
+	@Override
+	@SuppressWarnings("deprecation")
+	public ActionResultType use(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand hand, BlockRayTraceResult hit)
+	{
+		if(!player.isCrouching())
+		{
+			worldIn.setBlock(pos, state.setValue(POWER, state.getValue(POWER) == 0 ? 15 : 0), Constants.BlockFlags.NOTIFY_NEIGHBORS);
+			if(state.getValue(POWER) > 0)
+			{
+				worldIn.playSound(null, pos, SoundEvents.PISTON_EXTEND, SoundCategory.BLOCKS, 0.5F, 1.2F);
+				worldIn.getBlockTicks().scheduleTick(new BlockPos(pos), this, tickRate);
+			} else
+			{
+				worldIn.playSound(null, pos, SoundEvents.PISTON_CONTRACT, SoundCategory.BLOCKS, 0.5F, 1.2F);
+				worldIn.getBlockTicks().scheduleTick(new BlockPos(pos), this, tickRate);
+			}
+			
+			return ActionResultType.SUCCESS;
+		}
+		
+		return ActionResultType.PASS;
+	}
+	
+	@Override
+	public void tick(BlockState state, ServerWorld worldIn, BlockPos pos, Random rand)
+	{
+		super.tick(state, worldIn, pos, rand);
+		
+		if(state.getValue(POWER) >= 1)
+		{
+			worldIn.setBlockAndUpdate(pos, state.setValue(POWER, state.getValue(POWER) - 1));
+			worldIn.getBlockTicks().scheduleTick(new BlockPos(pos), this, tickRate);
+			
+			if(worldIn.getBlockState(pos).getValue(POWER) == 0)
```

Comment:
You shouldn't need to get the block state that you just set if you keep track of the power with a variable.